x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/__mocks__/mock_rules_table_persistent_state.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useKibana } from '../../../../../../common/lib/kibana';
+import { useGetInitialUrlParamValue } from '../../../../../../common/utils/global_query_string/helpers';
+import type { RulesTableSavedState } from '../rules_table_saved_state';
+
+export function mockRulesTablePersistedState({
+  urlState,
+  storageState,
+}: {
+  urlState: RulesTableSavedState | null;
+  storageState: RulesTableSavedState | null;
+}): void {
+  (useGetInitialUrlParamValue as jest.Mock).mockReturnValue(
+    jest.fn().mockReturnValue({ decodedParam: urlState })
+  );
+  (useKibana as jest.Mock).mockReturnValue({
+    services: { sessionStorage: { get: jest.fn().mockReturnValue(storageState) } },
+  });
```

Comment:
Not sure about that. This code will completely replace what mocked `useKibana` returns by default which could break a test if it will depend on both session storage and some other service.

I'd suggest to try mocking session storage in `x-pack/plugins/security_solution/public/common/lib/kibana/__mocks__/index.ts` and here it should retrieve the mocked session storage and call `services.sessionStorage.get.mockReturnValue(storageState)`.

Reply:
Correct me if I'm wrong I think you mean spying on items exposed by `useKibana` manual mock which documented [here](https://jestjs.io/docs/es6-class-mocks#spying-on-methods-of-our-class). The problem [it doesn't work](https://stackoverflow.com/questions/62369378/cannot-spy-on-mocked-functions-of-a-class-using-a-manual-mock). If I expose a spy try to mock return value or expect a number of calls it just doesn't catch the necessary calls like a two different instances are created.